ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -388,8 +403,8 @@ impl<'a> TraceBuilder<'a> {
             TraceAction::Promotion => todo!(),
         };
 
-        let first_blk = self.lookup_aot_block(&prev);
-        self.create_trace_header(self.aot_mod.block(&first_blk.unwrap()))?;
+        self.cp_block = self.lookup_aot_block(&prev);
+        self.create_trace_header(self.aot_mod.block(self.cp_block.as_ref().unwrap()))?;
```

Comment:
Can the `unwrap` fail?

Reply:
Probably, but that would mean something has seriously gone wrong in the tracer that it has given us a mappable block that doesn't exist in the AOT module.

Reply:
That's OK: let's document why we don't think it can fail.

Reply:
[4917a43](https://github.com/ykjit/yk/pull/998/commits/4917a4392d05659ed3841156ab696cc502997939)